src/sources/prometheus/scrape.rs

```diff
@@ -90,6 +93,37 @@ impl SourceConfig for PrometheusScrapeConfig {
             .endpoints
             .iter()
             .map(|s| s.parse::<http::Uri>().context(sources::UriParseSnafu))
+            .map(|r| {
+                if let Ok(uri) = r {
```

Comment:
Is this the only way to deal with `Result` objects being moved around?
As seen in the code, the best way I could find is the construction:
```rust
if let Ok(binding) = r {
  Ok(binding)
} else {
  r
}
```
Which is "okay" but not sure if there is a more idiomatic way.